cbor.js

```diff
@@ -50,12 +104,43 @@ function encode(value) {
     lastLength = length;
     return dataView;
   }
+
   function commitWrite() {
     offset += lastLength;
   }
-  function writeFloat64(value) {
-    commitWrite(prepareWrite(8).setFloat64(offset, value));
+
+  function encodeFloat16(value) {
+    var dataView = prepareWrite(3);
+    dataView.setUint8(offset, 0xf9);
+    dataView.setUint16(offset+1, getFloat16(value));
+    commitWrite(3);
+  }
+  
+  function encodeFloat(value) {
+    var dataView;
+    var f16 = checkFloat16(value);
+    if (f16 !== false) {
+      dataView = prepareWrite(3);
+      dataView.setUint8(offset, 0xf9);
+      dataView.setUint16(offset+1, f16);
+      commitWrite();
+      return;
+    }
+    var f32 = checkFloat32(value);
+    if (f32 !== false) {
+      dataView = prepareWrite(5);
+      dataView.setUint8(offset, 0xfa);
+      dataView.setUint32(offset+1, f32);
+      commitWrite();
+      return;
+    }
+    dataView = prepareWrite(9);
+    dataView.setUint8(offset, 0xfb);
+    dataView.setFloat64(offset+1, value);
+    commitWrite();
   }
```

Comment:
the encodeFloat behavior is different from the writeFloat behavior in that the encoding prefix is also written.
encodeFloat32 would not be used; checkFloat32 returns an integer, and written directly
encodeFloat16 is not used here; checkFloat16 returns an integer, and written directly
writeFloat64 could be used here, but would only be used here, and it's behavior is different, so it was removed